appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/DefaultFirebaseAppCheck.java

```diff
@@ -146,44 +176,58 @@ public void removeAppCheckListener(@NonNull AppCheckListener listener) {
   @NonNull
   @Override
   public Task<AppCheckTokenResult> getToken(boolean forceRefresh) {
-    if (!forceRefresh && hasValidToken()) {
-      return Tasks.forResult(DefaultAppCheckTokenResult.constructFromAppCheckToken(cachedToken));
-    }
-    if (appCheckProvider == null) {
-      return Tasks.forResult(
-          DefaultAppCheckTokenResult.constructFromError(
-              new FirebaseException("No AppCheckProvider installed.")));
-    }
-    // TODO: Cache the in-flight task.
-    return fetchTokenFromProvider()
-        .continueWithTask(
-            new Continuation<AppCheckToken, Task<AppCheckTokenResult>>() {
-              @Override
-              public Task<AppCheckTokenResult> then(@NonNull Task<AppCheckToken> task) {
-                if (task.isSuccessful()) {
-                  return Tasks.forResult(
-                      DefaultAppCheckTokenResult.constructFromAppCheckToken(task.getResult()));
-                }
-                // If the token exchange failed, return a dummy token for integrators to attach in
-                // their headers.
-                return Tasks.forResult(
-                    DefaultAppCheckTokenResult.constructFromError(
-                        new FirebaseException(
-                            task.getException().getMessage(), task.getException())));
-              }
-            });
+    return retrieveStoredTokenTask.continueWithTask(
+        new Continuation<Void, Task<AppCheckTokenResult>>() {
```

Comment:
nit: can you switch to lambda syntax here and below?

Reply:
Done.